dml_events.go

```diff
@@ -421,8 +421,10 @@ func appendEscapedString(buffer []byte, value string) []byte {
 	return append(buffer, '\'')
 }
 
-func appendEscapedBuffer(buffer, value []byte, binary bool) []byte {
-	if binary {
+func appendEscapedBuffer(buffer, value []byte, isJSON bool) []byte {
+	if isJSON {
+		buffer = append(buffer, "CAST("...)
+	} else {
 		buffer = append(buffer, "_binary"...)
```

Comment:
Yes it is only used for binary ([]byte). The flag previously was only used for JSON, which turns out to be incorrectly implemented.

Reply:
> Yes it is only used for binary ([]byte). The flag previously was only used for JSON, which turns out to be incorrectly implemented.

Are we certain that `[]byte` was only ever used for binary or JSON data? and that `string`, `[]byte`, `bool`, `float64/32` and `decimal.Decimal` cover all data types?

Reply:
After looking at the code base for a short period of time, I can't say for certain if all mysql types is converted to []byte. If there are any non blob and json types converted to []bytes, we definitely don't use it internally.

Since we are currently blocked by this patch, I'm going to merge it for now. I can do a more comprehensive survey once we unblock our internal mover.